playwright-specflow-xunit/Steps/StepDefinitions.cs

```diff
@@ -0,0 +1,88 @@
+﻿using Microsoft.Playwright;
+using Newtonsoft.Json;
+
+namespace  SpecFlowPlaywrightXUnitExample.Steps
+{
+    [Binding]
+    public class PlaywrightTestParallel
+    {
+        private IBrowser? _browser;
+        private Dictionary<string, object>? _capabilities = new Dictionary<string, object>();
+        private string gdpUrl = "";
+        private string currentUrl;
+
+        [BeforeScenario]
+        private async void CreateCapabilities()
+        {
+            string user, accessKey;
+            user = Environment.GetEnvironmentVariable("LT_USERNAME");
+            accessKey = Environment.GetEnvironmentVariable("LT_ACCESS_KEY");
+            
+            Dictionary<string, string> ltOptions = new Dictionary<string, string>();
+
+            ltOptions.Add("name", "Playwright Test");
+            ltOptions.Add("build", "Playwright C-Sharp XUnit tests");
+            ltOptions.Add("platform", "Windows 10");
+            ltOptions.Add("user", user);
+            ltOptions.Add("accessKey", accessKey);
+            
+            _capabilities.Add("browserVersion", "latest");
+            _capabilities.Add("LT:Options", ltOptions);
+        }
+        
+        [AfterScenario]
+        private async void CloseDriver()
+        {
+            await _browser.CloseAsync();
+        }
+
+        [Given(@"Set (.*) as a capability")]
+        public void CreateBrowser(string browserType)
+        {
+            _capabilities.Add("browserName", browserType);
+            string capabilitiesJson = JsonConvert.SerializeObject(_capabilities);
+            gdpUrl = "wss://cdp.lambdatest.com/playwright?capabilities=" + Uri.EscapeDataString(capabilitiesJson);
+        }
+    
+        [Then(@"Go to Bing")]
+        public async Task SearchWithBing()
+        {
+            using var playwright = await Playwright.CreateAsync();
+            _browser = await playwright.Chromium.ConnectAsync(gdpUrl);
+            var page = await _browser.NewPageAsync();
+            await page.GotoAsync("https://www.bing.com");
+            currentUrl = page.Url;
+
+            if (currentUrl.Contains("bing"))

```

Comment:
Can we use assertions or expect statements here rather than if statement and pass the error(if any) in the setTestStatus's remark param?

Reply:
If we put like that `Assert.True(currentUrl.Contains("bing"))` and if the assertion failed, we cannot send the result in the same method it will throw assertion exception and run the after method. So we can set test is failed in the after method but when we are setting the test failed we need `page` object. I tried to set page object global but somehow most probably because of async programming, it gives somewhere null pointer. That is why I kept it like that

Reply:
@alisengul-lambdatest , cant we create the test in before scenario via the params? As we can get the config required for a specific scenario in the beforeScenario block too.

Reply:
As far as I know we cannot give params to the BeforeMethod from feature file. It is running before scenario started. But maybe we can I don't know actually. @ankur-lt

Reply:
@mukesh-lt , can you verify and resolve this?